index/segment/mem/types.go

```diff
@@ -50,15 +42,48 @@ type Options interface {
 	// InstrumentOptions returns the instrument options.
 	InstrumentOptions() instrument.Options
 
-	// SetInsertQueueSize sets the insert queue size.
-	SetInsertQueueSize(value int64) Options
+	// SetPostingsListPool sets the PostingsListPool.
+	SetPostingsListPool(value segment.PostingsListPool) Options
+
+	// PostingsListPool returns the PostingsListPool.
+	PostingsListPool() segment.PostingsListPool
+
+	// SetInitialCapacity sets the initial capacity.
+	SetInitialCapacity(value int) Options
+
+	// InitialCapacity returns the initial capacity.
+	InitialCapacity() int
+}
 
-	// InsertQueueSize returns the insert queue size.
-	InsertQueueSize() int64
+// termDictionary represents a mapping from doc.Field -> doc
+type termsDictionary interface {
+	// Insert inserts a mapping from field `f` to document ID `docID`.
+	Insert(f doc.Field, docID segment.DocID) error
 
-	// SetNumInsertWorkers sets the number of insert workers.
-	SetNumInsertWorkers(value int) Options
+	// Fetch returns all the docIDs matching the given filter.
+	// NB(prateek): the returned PostingsList is safe to modify.
+	Fetch(fieldName []byte, fieldValueFilter []byte, isRegexp bool) (segment.PostingsList, error)
```

Comment:
Might be worth making the `isRegexp` field her an options struct.  I see there being more of these type of options in the future and might make cross-repo changes hard if its not a struct.

Reply:
Thoughts on this?

Reply:
+1 for the options struct since we may want to distinguish glob expressions from regexes at some point and who knows what else

Reply:
Sorry missed this earlier. Sounds good, will do.

Reply:
done